week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/Underground.ts

```diff
@@ -0,0 +1,52 @@
+import { Person } from './Person';
+
+export class Underground {
+  private static INITIAL_NO_OF_BARRICADES = 3
+  private noOfBarricades = Underground.INITIAL_NO_OF_BARRICADES
+
+  constructor(private prisoner: Person) {}
+
+  enter(savior: Person) {
+    // #RQ3: king cannot enter Underground !
+    if (savior.title === "king") {
+      throw new Error("king cannot enter Underground !")
+    } else {
```

Comment:
`else` nie jest potrzebne tutaj bo `throw` przerywa wykonywanie tej metody

Reply:
Trochę się zakopałem w modułach odnośnie baz danych i ORM. Wydaje mi się że jeśli będzie kolejna edycja tego szkolenia będę mógł bardziej skupić się na zadaniach domowych i dowieźć je do końca a nie zaczynać, tymczasem dzięki wielkie za review i mam nadzieję że zrobię to lepiej w następnej edycji